vignettes/epipredict.Rmd

```diff
@@ -1,470 +1,565 @@
-we wanted to actually predict the future. We could have instead used any test data
-that contained the necessary predictors.
+where `.pred` gives the point/median prediction, and `.pred_distn` is a
+`dist_quantiles()` object representing a distribution through various quantile
+levels.
+The `[6]` in the name refers to the number of quantiles that have been
+explicitly created[^4].
+By default, `.pred_distn` covers a 90% prediction interval, reporting the 5% and 95% quantiles.
+
+The `epi_workflow` is a significantly more complicated object, extending a
+`workflows::workflow()`  to include post-processing steps:
 
+```{r show_workflow}
+four_week_ahead$epi_workflow
+```
 
-## Conclusion
+An `epi_workflow()` consists of 3 parts:
+
+- `Preprocessor`: a collection of steps that transform the data to be ready for
+      modelling. Steps can be custom, as are those included in this package, 
+      or [be defined in `{recipes}`](https://recipes.tidymodels.org/reference/index.html).
+      `four_week_ahead` has 5 steps; you can inspect them more closely by
+      running `hardhat::extract_recipe(four_week_ahead$epi_workflow)`.[^6]
+- `Model`: the actual model that does the fitting, given by a
+  `parsnip::model_spec`. `four_week_ahead` uses the default of
+  `parsnip::linear_reg()`, which is a `{parsnip}` wrapper for
+  `stats::lm()`. You can inspect the model more closely by running
+  `hardhat::extract_fit_recipe(four_week_ahead$epi_workflow)`.
+- `Postprocessor`: a collection of layers to be applied to the resulting
+  forecast. Layers are internal to this package. `four_week_ahead` just so happens to have
+  5 of as these well. You can inspect the layers more closely by running
+  `epipredict::extract_layers(four_week_ahead$epi_workflow)`.
+
+See the [Custom Epiworkflows vignette](custom_epiworkflows) for recreating and then
+extending `four_week_ahead` using the custom forecaster framework.
+
+## Mathematical description
+
+Let's look at the mathematical details of the model in more detail, using a minimal version of
+`four_week_ahead`:
+
+```{r, four_week_again}
+four_week_small <- arx_forecaster(
+  covid_case_death_rates |> filter(time_value <= forecast_date),
+  outcome = "death_rate",
+  predictors = c("case_rate", "death_rate"),
+  args_list = arx_args_list(
+    lags = list(c(0, 7, 14), c(0, 7, 14)),
+    ahead = 4 * 7,
+    quantile_levels = c(0.1, 0.25, 0.5, 0.75, 0.9)
+  )
+)
+hardhat::extract_fit_engine(four_week_small$epi_workflow)
+```
 
-Internally, we provide some simple functions to create reasonable forecasts.
-But ideally, a user could create their own forecasters by building up the
-components we provide. In other vignettes, we try to walk through some of these
-customizations.
+If $d_t$ is the death rate on day $t$ and $c_t$ is the case rate, then the model
+we're fitting is:
 
-To illustrate everything above, here is (roughly) the code for the
-`flatline_forecaster()` applied to the `case_rate`.
+$$
```

Comment:
I don't like this expression. I would prefer either $E[d_{t+28} | d_t,...,d_{t-14}, c_t,
..., c_{t-14}]$ or as you have it with the addition of mean-0 noise
$\epsilon_t$. It probably also needs a $j$ subscript for geography.

Reply:
how's the rewrite? I went with the 2nd and added $j$ subscripts. Also a line break, since the $j$'s cause it to get a bit longer.